nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -29,30 +28,25 @@ class ServiceClient
 public:
   explicit ServiceClient(
     const std::string & service_name,
-    const rclcpp::Node::SharedPtr & provided_node = rclcpp::Node::SharedPtr())
-  : service_name_(service_name)
+    const rclcpp::Node::SharedPtr & provided_node)
+  : service_name_(service_name), node_(provided_node)
   {
-    if (provided_node) {
-      node_ = provided_node;
-    } else {
-      node_ = generate_internal_node(service_name + "_Node");
-    }
-    client_ = node_->create_client<ServiceT>(service_name);
-  }
-
-  ServiceClient(const std::string & service_name, const std::string & parent_name)
-  : service_name_(service_name)
-  {
-    node_ = generate_internal_node(parent_name + std::string("_") + service_name + "_client");
-    client_ = node_->create_client<ServiceT>(service_name);
+    callback_group_ = node_->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive,
+      false);
```

Comment:
This is me asking a question because I haven't used callback groups like this before (previously in the context of priorities only):

I see the second argument means whether to add it to the node's executor or not. Why do you need to run it against a separate executor if its in another callback group? Wouldn't the `MutuallyExclusive` callback group type make it so that it could be processed at the same time as other callbacks in the same executor?

Reply:
I've done some digging, is the reason that doesn't work because the executor being run right now is single threaded? So you're using a second single threaded executor to execute this callback group separately due to the lack of multithreaded executor? 

Edit: I've found that the default callback group when not set is MutuallyExclusive single threaded executor, so if you added this to the existing one, then it wouldn't work because it was already blocked. It seems odd to me to have an internal executor - but I think this is OK. Any significan toverhead associated with just an orphaned executor over here for just this object?

Reply:
The second argument will automatically add the callback group to the node executor.
Hence, first, the node needs an executor, and second, you need to change the way you spin your node.
```c++
executor.add_node(node);
executor.spin();
```

Reply:
Using `MutuallyExclusive`, only the callbacks in the same callback group are mutually exclusive. If you create a callback group per client, there are not blocking each other.